packages/otelbin/src/app/s/[id]/metadataUtils.test.ts

```diff
@@ -46,26 +46,77 @@ describe("extractComponents", () => {
 });
 
 describe("calcScale", () => {
-	it("Should calculate the correct scale based on the maximum height or width of the visualization nodes to ensure they fit within the standard 1200x630 size of the generated Open Graph image.", () => {
-		const nodes: Node[] = [
-			{ type: "parentNodeType", data: { height: 100 }, position: { x: 0, y: 0 }, id: "1" },
-			{ type: "parentNodeType", data: { height: 200 }, position: { x: 0, y: 0 }, id: "2" },
-			{ type: "processorsNode", data: { height: 100 }, position: { x: 0, y: 0 }, id: "3" },
-			{ type: "processorsNode", data: { height: 100 }, position: { x: 0, y: 0 }, id: "4" },
-		];
-
-		const edgeWidth = 10;
-		const scale = calcScale(edgeWidth, nodes);
-		expect(scale).toBe("1.270");
-	});
+	const edgeWidth = 10;
 
-	it("Should return 1 if there is no node.", () => {
-		const nodes: Node[] = [];
+	const runTest = (nodes: Node[], expectedScale: string) => {
+		it("Should calculate the correct scale based on the maximum height or width of the visualization nodes to ensure they fit within the standard 1200x630 size of the generated Open Graph image.", () => {
+			const scale = calcScale(edgeWidth, nodes);
+			expect(scale).toBe(expectedScale);
+		});
+	};
 
-		const edgeWidth = 10;
-		const scale = calcScale(edgeWidth, nodes);
-		expect(scale).toBe("1");
-	});
+	runTest(
```

Comment:
Good refactoring! For next time: Did you know Jest has native support for something like this? :)

https://github.com/dash0hq/otelbin/blob/main/packages/otelbin/src/lib/utils.test.ts#L9-L34